devito/operator/operator.py

```diff
@@ -148,46 +157,32 @@ def _build(cls, expressions, **kwargs):
         if any(not isinstance(i, Eq) for i in expressions):
             raise InvalidOperator("Only `devito.Eq` expressions are allowed.")
 
-        name = kwargs.get("name", "Kernel")
-        dse = kwargs.get("dse", configuration['dse'])
-
         # Python-level (i.e., compile time) and C-level (i.e., run time) performance
         profiler = create_profile('timers')
 
-        # Lower input expressions to internal expressions (e.g., attaching metadata)
+        # Lower input expressions
         expressions = cls._lower_exprs(expressions, **kwargs)
 
-        # Group expressions based on their iteration space and data dependences
-        # Several optimizations are applied (fusion, lifting, flop reduction via DSE, ...)
-        clusters = clusterize(expressions, dse_mode=set_dse_mode(dse))
+        # Group expressions based on iteration spaces and data dependences
+        clusters = cls._lower_clusters(expressions, **kwargs)
 
-        # Lower Clusters to a Schedule tree
-        stree = st_build(clusters)
+        # Lower Clusters to a ScheduleTree
+        stree = cls._lower_stree(clusters, **kwargs)
 
-        # Lower Schedule tree to an Iteration/Expression tree (IET)
-        iet = iet_build(stree)
-
-        # Instrument the IET for C-level profiling
-        iet = profiler.instrument(iet)
-
-        # Wrap the IET with a Callable
-        parameters = derive_parameters(iet, True)
-        op = Callable(name, iet, 'int', parameters, ())
-
-        # Lower IET to a Target-specific IET
-        op, target_state = cls._specialize_iet(op, **kwargs)
+        # Lower ScheduleTree to an Iteration/Expression Tree
+        iet, byproduct = cls._lower_iet(stree, profiler, **kwargs)
 
         # Make it an actual Operator
-        op = Callable.__new__(cls, **op.args)
+        op = Callable.__new__(cls, **iet.args)
         Callable.__init__(op, **op.args)
 
         # Header files, etc.
         op._headers = list(cls._default_headers)
-        op._headers.extend(target_state.headers)
+        op._headers.extend(byproduct.headers)
```

Comment:
My VScode pylint says
`Instance of 'Callable' has no '_headers' member` on this line?
SHould we care?

Reply:
Maybe some info about byproduct? Like a docstring?

Reply:
Maybe what I say its driven form the fact that byproduct is a bit vague.

Reply:
> My VScode pylint 

it's buggy. `op` is of type Operator

> Maybe some info about byproduct? Like a docstring?

I could..... but it feels like an overkill. `byproduct` is simply the by-product of `lower-iet` which is a bag of additional objects and properties